Public/Src/Utilities/Storage/ChangeTracking/FileChangeProcessor.cs

```diff
@@ -14,21 +15,26 @@ namespace BuildXL.Storage.ChangeTracking
     public class FileChangeProcessor
     {
         private readonly FileChangeTracker m_fileChangeTracker;
-        private ScanningJournalResult m_scanningJournalResult;
-        private readonly List<FileChangeTrackerUnsubscriber> m_unsubscribers = new List<FileChangeTrackerUnsubscriber>(2);
+        private readonly InputChangeList m_inputChangeList;
+        private readonly List<FileChangeTrackerUnsubscriber> m_fileChangeTrackerUnsubscribers = new List<FileChangeTrackerUnsubscriber>(2);
+        private readonly List<InputChangeListUnsubscriber> m_inputChangeListUnsubscribers = new List<InputChangeListUnsubscriber>(2);
         private readonly LoggingContext m_loggingContext;
 
         /// <summary>
         /// Creates an instance of <see cref="FileChangeProcessor"/>.
         /// </summary>
-        public FileChangeProcessor(LoggingContext loggingContext, FileChangeTracker fileChangeTracker)
+        public FileChangeProcessor(
+            LoggingContext loggingContext, 
+            FileChangeTracker fileChangeTracker, 
+            InputChangeList inputChangeList = null)
         {
             Contract.Requires(loggingContext != null);
             Contract.Requires(fileChangeTracker != null);
             Contract.Requires(fileChangeTracker.IsTrackingChanges);
 
             m_loggingContext = loggingContext;
             m_fileChangeTracker = fileChangeTracker;
+            m_inputChangeList = inputChangeList;

```

Comment:
Shouldn't it be either the USN change tracker -or- the input file? Or will se not trust this and union them anyway....

Reply:
The primary one is always the USN, and is handled by file change tracker. The input change list is simply additions to force pips to dirty. For example, in a single build, there may be no changes to the FS, but `git status` may shows changes wrt. the source control.

---
In reply to: [313496685](https://github.com/microsoft/BuildXL/pull/726#discussion_r313496685) [](ancestors = 313496685)